### PR TITLE
[-] Command Generate - Now generate most of the many to many relationships

### PR DESCRIPTION
**Pull request by VincentMolinie**

### CODE REVIEW

**Comment by VincentMolinie** [comment]
[Lumber - Generate more ManyToMany](https://trello.com/c/LvH65zJY/3987-lumber-generate-more-manytomany)

File: services/analyzer/sequelize-tables-analyzer.js [hunk 1]
```diff
@@ -141,86 +163,83 @@ async function analyzeTable(table, config) {
...
+
+  Object.keys(databaseSchema).forEach((tableName) => {
+    const table = databaseSchema[tableName];
     const { constraints, primaryKeys } = table;
-    constraints.forEach((constraint) => {
+    const { isJunction } = schemaGenerated[tableName].options;
+
+    const foreignKeysWithExistingTable = constraints
+      .filter((constraint) => constraint.columnType === 'FOREIGN KEY'
+        && databaseSchema[constraint.foreignTableName]);
```
Comment (arnaudbesnier) [inline services/analyzer/sequelize-tables-analyzer.js]: Is it a fix/protection compared to the current implementation?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 - Technical - Reorganize test utils/fixtures/expect folders.
 - Technical - Set the latest `forest-express` dependencies version while generating a project.
 - Technical - Reorder in alphabetical order `options` inside `db-analysis-output`.
+- Command Generate - Now generate most of the many to many relationships.
```
Comment (arnaudbesnier) [inline CHANGELOG.md]: You should declare precisely the new way many-to-many associations are detected.

File: services/analyzer/sequelize-tables-analyzer.js [hunk 3]
```diff
@@ -85,6 +85,28 @@ function hasIdColumn(fields, primaryKeys) {
     || _.includes(primaryKeys, 'id');
 }
 
+function isJunctionTable(fields, constraints) {
+  const FIELDS_TO_IGNORE = [
+    'createdAt', 'updatedAt', 'deletedAt',
+    'creationDate', 'updateDate', 'deleteDate',
+  ];
```
Comment (arnaudbesnier) [inline services/analyzer/sequelize-tables-analyzer.js]: The way we detect technical timestamps is error prone IMO.
We don't even check the `type` of the columns.

File: services/analyzer/sequelize-tables-analyzer.js (hunk 1 above)
Comment (VincentMolinie) [reply services/analyzer/sequelize-tables-analyzer.js]: It's a protection and a fix a possible issues on cross schema. It should be in another PR I forgot to move it. Do you still want me to move it ?

File: services/analyzer/sequelize-tables-analyzer.js (hunk 3 above)
Comment (VincentMolinie) [reply services/analyzer/sequelize-tables-analyzer.js]: I don't think so but tell me in which case it could ?

File: CHANGELOG.md (hunk 2 above)
Comment (VincentMolinie) [reply CHANGELOG.md]: Better ?

File: services/analyzer/sequelize-tables-analyzer.js (hunk 1 above)
Comment (arnaudbesnier) [reply services/analyzer/sequelize-tables-analyzer.js]: No but you should add a change description in the changelog if you think this was an issue in the previous implementation.

File: CHANGELOG.md (hunk 2 above)
Comment (arnaudbesnier) [reply CHANGELOG.md]: Did you change something?

File: services/analyzer/sequelize-tables-analyzer.js (hunk 3 above)
Comment (arnaudbesnier) [reply services/analyzer/sequelize-tables-analyzer.js]: you cannot anticipate all the technical timestamps name you can have (coming from custom database column names, from ORM migrations, from database conventions...)
For instance `touchAt`, `touchedAt`, `cDate`, `cAt`, `created`, `dateCreation`,... could be technical timestamp too.

Never mind, we'll iterate on this if issues occur.

**Pull request merged**